### PR TITLE
Introduce provider/ec2/internal/ec2instancetypes

### PR DESCRIPTION
**Pull request by axw**
This diff adds a new internal package to the ec2
provider to contain instance type information.
The data is generated from the "Price List API":
    http://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/price-changes.html

Included is a command that will be run by
"go generate". You must download the index.json
file from https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/index.json,
and then "go generate" will write generated.go.

In the future we may wish to extract functionality
from the "process_cost_data.go" command, so that
we can refresh the instance types and cost data
at runtime. This should not be difficult to do,
but it is beyond the scope of this diff.

The output has been compared to the existing hard-
coded instance type data in provider/ec2. There
are some differences:
 - the "cpu-power" values for the new data are
   not completely correct. This is because the
   price list API does not include that information.
   These values are estimated based on a rough
   formula involving the vCPUs and clock speed.
 - some of the old data was out-dated
 - there is information for regions that we did
   not previously include (namely ap-south-1)
 - some instance types are now considered HVM
   whereas they were PV before. For example, the
   M3 instance types are capable of both PV and
   HVM, but we previously said they were PV. We
   *should* allow either, but in lieu of doing
   that, we prefer HVM as Amazon suggests.

### CODE REVIEW

**Comment by axw** [comment]
**QA**

- I have compared the output to existing instance type data, and found no problems apart from mildly inaccurate cpu-power values. That should not matter, as it's only supposed to be a rough estimate of CPU power anyway.
- With another WIP branch, I have bootstrapped using the new instance type information.

File: provider/ec2/internal/ec2instancetypes/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+index.json
```
Comment (howbazaar) [inline provider/ec2/internal/ec2instancetypes/.gitignore]: Why is this here rather than at the top level? I don't suppose it matters, just was surprising to see.

**Comment by axw** [comment]
> I'd really like a doc.go file that explains how to refresh the generated information rather than just a few lines of comment that could easily be missed.

I've added a doc.go.

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$
